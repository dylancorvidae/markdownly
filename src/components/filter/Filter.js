import React from 'react';
import PropTypes from 'prop-types';

const Filter = ({ handleSubmit, handleChange, searchTerm }) => {
  return (
    <form onSubmit={handleSubmit}>
      <input onChange={handleChange}></input>
      <button onClick={() => handleSubmit(searchTerm)}>Search</button>
    </form>

  );
};

Filter.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  searchTerm: PropTypes.string
};

export default Filter; 
