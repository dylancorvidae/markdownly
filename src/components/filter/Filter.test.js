import React from 'react';
import { shallow } from 'enzyme';
import Filter from './Filter';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Filter handleSubmit={() => {}} handleChange={() => {}} searchTerm={'hello'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
