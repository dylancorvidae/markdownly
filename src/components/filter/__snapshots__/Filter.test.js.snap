// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders Filter 1`] = `
<form
  className="Filter"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
